common/wuanalysis/anarule.cpp

```diff
@@ -129,20 +129,49 @@ class IoSkewRule : public AActivityRule
     {
         stat_type ioAvg = activity.getStatRaw(stat, StAvgX);
         stat_type ioMaxSkew = activity.getStatRaw(stat, StSkewMax);
+        unsigned actkind = activity.getAttr(WaKind);
         if (ioMaxSkew > options.queryOption(watOptSkewThreshold))
         {
             stat_type timeMaxLocalExecute = activity.getStatRaw(StTimeLocalExecute, StMaxX);
             stat_type timeAvgLocalExecute = activity.getStatRaw(StTimeLocalExecute, StAvgX);
 
             stat_type cost;
-            //If one node didn't spill then it is possible the skew caused all the lost time
-            unsigned actkind = activity.getAttr(WaKind);
-            if ((actkind==TAKspillread||actkind==TAKspillwrite) && activity.getStatRaw(stat, StMinX) == 0)
+            if ((actkind==TAKspillread||actkind==TAKspillwrite) && (activity.getStatRaw(stat, StMinX) == 0))
+            {
+                //If one node didn't spill then it is possible the skew caused all the lost time
                 cost = timeMaxLocalExecute;
+                result.set(ANA_IOSKEW_RECORDS_ID, cost, "Uneven worker spilling is causing uneven %s time", category);
+            }
             else
+            {
+                bool sizeSkew = false;
+                bool numRowsSkew = false;
+                IWuEdge *wuEdge = nullptr;
+                if ((stat==StTimeDiskWriteIO) || (actkind==TAKspillwrite))
+                {
+                    if (activity.getStatRaw(StSizeDiskWrite)>options.queryOption(watOptSkewThreshold))
```

Comment:
testing the wrong stat.  Needs to be getStatRaw(StSizeDiskWrite, StSkewMax).  Same for the code below.